libraries/libfc/include/fc/crypto/common.hpp

```diff
@@ -42,10 +43,30 @@ namespace fc { namespace crypto {
       }
    };
 
+   // for specializations that do not meet SerializableForMemcmp concept
    template<typename T>
-   struct less_comparator {
+   struct less_comparator {};
+
+   template<typename T>
+   concept SerializableForMemcmp = requires(const T& obj) {
+      { obj.serialize() } -> std::ranges::forward_range;
+      { obj.serialize().data() } -> std::contiguous_iterator; // Ensures data is contiguous for memcmp
+      { obj.serialize().size() } -> std::convertible_to<std::size_t>;
+   } && std::is_trivially_copyable_v<std::ranges::range_value_t<decltype(std::declval<const T&>().serialize())>>;
```

Comment:
Nice.  Should/could this be generified and moved to something like `(key|crypto)_traits.hpp`?